GDJS/Runtime/AsyncTasksManager.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * GDevelop JS Platform
+ * Copyright 2013-2022 Florian Rival (Florian.Rival@gmail.com). All rights reserved.
+ * This project is released under the MIT License.
+ */
+namespace gdjs {
+  /**
+   * A manager for all asynchronous tasks.
+   * @see {@link RuntimeScene.getAsyncTasksManager}.
+   */
+  export class AsyncTasksManager {
+    /**
+     * Maps a task to the callback to be executed once it is finished.
+     */
+    private tasks = new Map<AsyncTask, (runtimeScene: RuntimeScene) => void>();
```

Comment:
I think it's indeed better not to store the callback inside the AsyncTask, because we don't want to run the risk of a rogue/bad AsyncTask running its callback (it's the job of processTasks).

This being said, what do you think of just storing an array of `{asyncTask: AsyncTask, callback: (RuntimeScene) => void}`?
A Map is something more complex and iterating on the keys involves for the JS engine to build an iterator. Iterating on an array on the contrary is a simple pointer increment, which might be much more efficient when compiled by JS engines.

The disadvantage is that removing an element in an array though is that it's in the worst case O(n) (n being the length). Maybe we could use a Linked List to iterating quickly (though this is less cache friendly, because you have to jump from pointers to pointers) and delete quickly (just two references to change).
Or we use a simple array, iterate on all tasks and only filter the array at the end.

Just some food for thought on the data structures :)